### DISCUSSION TITLE
BBC Micro Emulator

### DISCUSSION THREAD

**Discussion by DrBuns3n** [eduardofilo/RG350_adam_image#93] [Ideas & Requests]
The RG350 has probably one of the best BBC Micro emulators on any platform, based on Beebem. The author's github is below. Any chance of implementing this in the Adam image?

https://github.com/jeremyrayner/beebem-rg350

**Comment by SuperBadger77** [comment]
I'd love the Beebem emulator to be included as well. It's my favourite computer / emulator. I've manually added it in myself to the Adam image but the current version doesn't seem compatible with OD Beta. I've tried contacting the owner of the link you posted but not had a response.

**Comment by eduardofilo** [comment]
@SuperBadger77 your avatar gives yourself away ;-)
Well, it will be necessary to see if that emulator can be made compatible with OD Beta. I will take a look.

**Comment by SuperBadger77** [comment]
:-)

**Comment by DrBuns3n** [comment]
Keeping my fingers crossed! Good luck and thanks for looking into this @eduardofilo

**Comment by DrBuns3n** [comment]
FYI @plrguez fixed beebem for OD beta, so I think their version should work on the Adam image?: 

https://github.com/plrguez/beebem-od/releases/tag/0.6.2-od-2021-10-03-fix01

**Comment by SuperBadger77** [comment]
Bloody superb! Will test it out.

**Comment by SuperBadger77** [comment]
Tried this on my RG280V. Works perfectly. Please can we get this added to the Adam build?

**Comment by SuperBadger77** [comment]
> FYI @plrguez fixed beebem for OD beta, so I think their version should work on the Adam image?: 
> 
> 
> 
> https://github.com/plrguez/beebem-od/releases/tag/0.6.2-od-2021-10-03-fix01
> 
> 

Thanks for finding this. Made my day / week / month. Now all we need is a version that works on the RK3326 chipset 😀

**Comment by eduardofilo** [comment]
Well, in that case, I'll incorporate it into the image, yes.

**Comment by plrguez** [comment]
Hi Eduardo

I am working to fix the issue with flickering colors appearing on the screen on some devices. The current version does not guarantee that this has been solved.
 
I observed the problem on RG280M but not in RG350P after fix release. I will test new releases also on RG350M and on LDKv when I built for Lepus OD beta. My suspiction is that only 320x480 panels are affected with this problem.

So if you include beebem with the current fix in Adam, I think that you should warn users about this potential problem. Or you can wait until we are sure the problem is gone. And I hope the solution can see the light soon :)

Greetings and congratulations on the excellent work on Adam 

@SuperBadger77 if you notice this problem please let me know, because I believe the RG280V is one of the devices potentially affected by this problem.

**Comment by SuperBadger77** [comment]
@plrguez 10 4 - will try it on a few more devices and report back. Thank you so much :-)

**Comment by eduardofilo** [comment]
Understood @plrguez and thanks for your work too.
I have a RG280V also, so I will make test and if anything is found I will tell you.

**Comment by SuperBadger77** [comment]
Tried it in an RG300X and it works fine. I'll carry on with the testing. @plrguez don't suppose you'd be interested in porting the emulator over to the RG351 series / Rk3266 chipset? 😄

**Comment by plrguez** [comment]
Sorry, I don't have an RG351 for now and I will not have in short term.

By the way I have just released another build for OD beta. I have added hardware scaling (IPU), lepus build and some transparency to virtual keyboard. See it [here](https://github.com/plrguez/beebem-od/releases/tag/0.6.3-od-2021-10-09-fix02)

**Comment by SuperBadger77** [comment]
Great work @plrguez if you need me to do any testing then please let me know. Think it's ready for the next Adam release :-) If you let me know how I'd compile and get Beebem on RG351 devices I could have a crack it myself? My coding skills are old and out of date but I'm willing to give it a go.

**Comment by eduardofilo** [comment]
I have integrated BeebEm in the next Adam version. Be noticed that I have redirected the `roms` (BIOS) folder to `bios` directory in EXT SD card, like all other emulators. The ROMs directory in SimpleMenu will be `roms/BBCMICRO`. All this details will be available in README.

**Comment by SuperBadger77** [comment]
Fantastic news :-) I shall test it as soon as it is released. For the missing themes in SimpleMenu - it shouldn't crash, it just won't show anything. I hacked together some that worked for Epic Noir 320 and 480 screen sizes. I'll look at the best ways of getting thes incorporated into the themes or see if it is possible the owners will update to add it in. Any advice \ input appreciated. Thank you to everyone in getting this to work - much much appreciated :-)

**Comment by eduardofilo** [comment]
0A and ComicBook already had resources for BBC Micro. In BigCoy I have put an image of code to indicate the educational side of the computer. SimUI being text-based it is not difficult to adapt.

![logo](https://user-images.githubusercontent.com/2923728/136796096-12098d5d-ad1a-475b-8276-d0d54ea3f0fd.png)
.

**Comment by plrguez** [comment]
> Great work @plrguez if you need me to do any testing then please let me know. Think it's ready for the next Adam release :-) If you let me know how I'd compile and get Beebem on RG351 devices I could have a crack it myself? My coding skills are old and out of date but I'm willing to give it a go.

I don't know anything about RG351 development, but if it have an GCC based toolchain and support SDL 1.2 for viedo and audio then probably with a few adjusts can be built for them. The `Makefile_RG350` file is used to build for OpenDingux. The flags need to be adjusted because there are some of them for MIPS architecture. The IPU scaler probably don't work, but I don't sure. The key/buttons processing for sure is need to be changed. The code I have introduced use the OPENDINGUX directive.

**Comment by eduardofilo** [comment]
BBC Micro showed up in 1.3 release. :-)
